kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/admin/ProcessAdminServiceBase.java

```diff
@@ -90,6 +99,35 @@ public MigrationReportInstanceList migrateProcessInstances(String containerId, L
         return convertMigrationReports(reports);
     }
 
+    public MigrationReportInstanceList migrateProcessInstanceWithAllSubprocess(String containerId,
+                                                                               Long processInstanceId,
+                                                                               String targetContainerId,
+                                                                               String targetProcessId,
+                                                                               String payload, 
+                                                                               String marshallingType) {
+
+        ProcessInstanceDesc pi = runtimeDataService.getProcessInstanceById(processInstanceId);
+        if(pi.getParentId() > 0) {
+            throw new IllegalArgumentException("Only root process can invoke this migration with all subprocesses");
+        }
+        List<Long> processInstancesId =  runtimeDataService.getProcessInstancesWithSubprocessByProcessInstanceId(processInstanceId, 
+                                                                                                                 asList(ProcessInstance.STATE_ACTIVE), 
+                                                                                                                 new QueryContext(0, -1)).stream().map(ProcessInstanceDesc::getId).collect(Collectors.toList());
+
+        Map<String, String> nodeMapping = emptyMap();
+        if (payload != null) {
+            logger.debug("About to unmarshal node mapping from payload: '{}' using container {} marshaller", payload, containerId);
+            nodeMapping = marshallerHelper.unmarshal(containerId, payload, marshallingType, Map.class);
+        }
+
+        logger.debug("About to migrate process instances with ids {} from container '{}' to container '{}' and process id '{}' with node mapping {}",
+                processInstancesId, containerId, targetContainerId, targetProcessId, nodeMapping);
+        List<MigrationReport> reports = processInstanceMigrationService.migrate(containerId, convert(processInstancesId), targetContainerId, targetProcessId, nodeMapping);
+
+        logger.debug("Migration of process instances {} finished with reports {}", processInstancesId, reports);
+        return convertMigrationReports(reports);
```

Comment:
To avoid code duplication, we could call `migrateProcessInstances` here instead. 
```suggestion
        return migrateProcessInstances(containerId, processInstancesId, targetContainerId, targetProcessId, payload, marshallingType);
```

Reply:
done